scripts/prohibit-class-name-hardcode-in-strings

```diff
@@ -8,6 +8,6 @@ source "$self_dir"/utils.sh
 class_names=$(git ls-files '*.java' | xargs basename -s .java | sort | paste -sd'|')
 
 # Let's allow for class names preceded with `action.GitMachete.` (for property keys).
-if git grep -PHn '^[^"]*"[^"$]*(?<!action\.GitMachete\.)\b('"$class_names"')\b' -- '*.java' ':!frontendDefs' ':!frontendResourceBundles'; then
+if git grep -PHn '^[^"]*"[^"$]*(?<!(action\.GitMachete\.|string\.))\b('"$class_names"')\b' -- '*.java' ':!frontendDefs' ':!frontendResourceBundles'; then
```

Comment:
This is crashing, lookbehind assertion must be fixed length in almost all modern regex engines

Reply:
Actually... `.GitMachete.` segment can be included in the `EnhancedGraphTable` keys for consistency as well

Reply:
`(?<!(action|string)\.GitMachete\.)` should be okay